### DISCUSSION TITLE
[General] Failed to bridge ERC-20 token on mainnet

### DISCUSSION THREAD

**Discussion by elek** [zkSync-Community-Hub/zksync-developers#121] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Hi, I am trying to deposit STORJ tokens (to create the L2 bridged contract), but without any success. 

When I followed https://era.zksync.io/docs/reference/concepts/bridging-asset.html (and do `yarn hardhat deploy-zksync --script deposit.ts`), I got:

```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transaction={"to":"0xB64ef51C888972c908CFacf59B47C1AfBC0Ab8aC","data":"0x06fdde03","accessList":null}, code=CALL_EXCEPTION, version=providers/5.7.2)
```

Which is weird, as `0x06fdde03` is `name()`, I don't know why is it called from a transaction. And hard to debug what's going on...

We also tried to create the transaction manually:

https://etherscan.io/tx/0x33c81aea51b66b4fa3b89503d626d499b4a3ec85b349bd987483eca45af00983

It is committed, and I received the remaining ETHs on L2, but [token contract](https://explorer.zksync.io/address/0xA0806DA7835a4E63dB2CE44A2b622eF8b73B5DB5#transactions)  is not created (I started with value=0, but in that case it's failed with `mv` error)

What is the easiest way to deposit a new ERC-20 token?

### What is the specific issue or error you're encountering?

See above. I couldn't bridge new ERC-20 token.

Using hardhat example: error (see above)

Using manual transaction: committed, but new bridge is not created...

### Can you share the error messages or logs you're receiving, if any?

Yes, I can. And I did. See above.

### Have you made any recent changes to the contract before encountering this issue?

no

### Are there any external libraries or contracts that your contract interacts with?

not relevant

### Can you provide the relevant portions of your contract code where the issue is occurring?

no contract on my side

### Have you tried to isolate the problem, and if so, what were the results?

yes I tried it with a manual transaction, see brief description.

### What steps have you already taken to try to resolve the issue?

See brief description

**Comment by MexicanAce** [comment]
Hi @elek 👋 

Could you please post the contract and `deposit.ts` script that you're running?

> Which is weird, as 0x06fdde03 is name(), I don't know why is it called from a transaction

Name is required for the correct deployment of the L2 counterpart contract. Make sure that you are using the correct L1 provider and it can call name function of the contract.

**Comment by elek** [comment]
`deposit.ts` is just the copy pasted example: 

As you see, I am trying to bridge `0xB64ef51C888972c908CFacf59B47C1AfBC0Ab8aC` which has a working `name()` method (returns with `StorjToken`)

```
import { Wallet, Provider } from "zksync-web3";
import * as ethers from "ethers";
import { HardhatRuntimeEnvironment } from "hardhat/types";
import { Deployer } from "@matterlabs/hardhat-zksync-deploy";

// load env file
import dotenv from "dotenv";
dotenv.config();

// HTTPS RPC endpoint
const MAINNET_RPC_ENDPOINT = "https://mainnet.infura.io/v3/.....";

// Token address
const TOKEN_ADDRESS = "0xB64ef51C888972c908CFacf59B47C1AfBC0Ab8aC";

// Amount of tokens
const AMOUNT = "5";

const WALLET_PRIV_KEY = process.env.WALLET_PRIV_KEY || "";

if (!WALLET_PRIV_KEY) {
  throw new Error("Wallet private key not configured in env file");
}

export default async function (hre: HardhatRuntimeEnvironment) {
  console.log(`Running script to bridge ERC20 to L2`);

  // Initialize the wallet.
  const provider = new Provider(
    // @ts-ignore
    MAINNET_RPC_ENDPOINT || hre.config.networks.zkSyncTestnet.ethNetwork
  );

  const wallet = new Wallet(WALLET_PRIV_KEY, provider);

  // Create deployer object
  const deployer = new Deployer(hre, wallet);

  // Deposit ERC20 tokens to L2
  const depositHandle = await deployer.zkWallet.deposit({
    to: deployer.zkWallet.address,
    token: TOKEN_ADDRESS,
    amount: AMOUNT, // assumes ERC20 has 18 decimals
  });

  console.log(`Deposit transaction sent ${depositHandle.hash}`);
  console.log(`Waiting for deposit to be processed in L2...`);
  // Wait until the deposit is processed on zkSync
  await depositHandle.wait();
  console.log(`ERC20 tokens available in L2`);
}
```

**Reply by jtolio** [reply]
@MexicanAce - we're hoping we can have this resolved soon. Any updates?

**Comment by vasyl-ivanchuk** [comment]
Hi @elek , please check [Deposit ERC20 tokens section](https://era.zksync.io/docs/reference/concepts/bridging-asset.html#deposit-erc20-tokens) in the same tutorial. The difference is that for ERC20 tokens you should specify `approveERC20: true` in the `deposit` function:

```
// Deposit ERC20 tokens to L2
  const depositHandle = await deployer.zkWallet.deposit({
    to: deployer.zkWallet.address,
    token: TOKEN_ADDRESS,
    amount: ethers.utils.parseEther(AMOUNT), // assumes ERC20 has 18 decimals
    // performs the ERC20 approve action
    approveERC20: true,
  });
```

does it help?

**Reply by elek** [reply]
No, it doesn't. I did the approval earlier.

The transaction what I shared shows that the STORJ tokens are indeed transferred from the wallet what I used to the bridge. The balance of the bridge is increased.

Also, I tried it in two different ways:
 * with the manual transaction the STORJ tokens are transferred to the bridge, but the contract is not created on L2. As the bridge was able to transferred, the approval shouldn't be a problem.
 * using the suggested ts code is also failed, the very cryptic error message is not related to the approval as far as I see.
 
 I just double checked and the allowance limit of the bridge is still 190 (where the spender is the bridge). (you can double check: the details in the transaction what I shared earlier).

**Comment by dutterbutter** [comment]
Hey all, 

This [issue](https://github.com/zksync-sdk/zksync2-js/issues/2) has been addressed and will be included in the next release. Thanks for reporting the issue!

**Reply by elek** [reply]
Thanks for the fix. Do you have any information why the L2 side of the ERC-20 bridge is not created as a result of transaction https://etherscan.io/tx/0x33c81aea51b66b4fa3b89503d626d499b4a3ec85b349bd987483eca45af00983 ?

**Reply by elek** [reply]
Opened a separated discussion for this part of the question: https://github.com/zkSync-Community-Hub/zkync-developers/discussions/159

**Reply by dutterbutter** [reply, marked as answer]
@elek release has been published [here](https://www.npmjs.com/package/zksync-web3/v/0.15.5). Here is a working script for simplicity: 

```
import { Wallet, Provider, utils } from "zksync-web3";
import * as ethers from "ethers";

// load env file
import dotenv from "dotenv";
dotenv.config();

// HTTP RPC endpoints
const L1_RPC_ENDPOINT = process.env.L1_RPC_ENDPOINT || "";  // or an RPC endpoint from Infura/Chainstack/QuickNode/etc.
const L2_RPC_ENDPOINT = process.env.L2_RPC_ENDPOINT || "https://testnet.era.zksync.dev/"; // or the zkSync Era mainnet 

// ERC-20 Token (DIA) address in L1
const TOKEN_ADDRESS = "0x5C221E77624690fff6dd741493D735a17716c26B";

// Amount of tokens 
const AMOUNT = "1";

const WALLET_PRIV_KEY = process.env.WALLET_PRIV_KEY || "";

if (!WALLET_PRIV_KEY ) {
  throw new Error("Wallet private key is not configured in env file");
}

if (!L1_RPC_ENDPOINT) {
  throw new Error(
    "Missing L1 RPC endpoint. Check chainlist.org or an RPC node provider"
  );
}

if (!TOKEN_ADDRESS ) {
  throw new Error("Missing address of the ERC-20 token in L1");
}

async function main() {
  console.log(`Running script to bridge ERC-20 to L2`);

  // Initialize the wallet.
  const l1provider = new Provider(L1_RPC_ENDPOINT);
  const l2provider = new Provider(L2_RPC_ENDPOINT);
  const wallet = new Wallet(WALLET_PRIV_KEY, l2provider, l1provider);

  console.log(`L1 Balance is ${await wallet.getBalanceL1()}`);
  console.log(`L2 Balance is ${await wallet.getBalance()}`);

  // Deposit token to L2
  const depositHandle = await wallet.deposit({
    to: wallet.address,  // can bridge to a different address in L2
    token: TOKEN_ADDRESS,
    amount: ethers.utils.parseEther(AMOUNT), // assumes ERC-20 has 18 decimals
    // performs the ERC-20 approve action
    approveERC20: true,
  });
  console.log(`Deposit transaction sent ${depositHandle.hash}`);
  console.log(`Please wait a few minutes for the deposit to be processed in L2`);
}

main().then().catch((error) => {
    console.error(error);
    process.exitCode = 1;
});
```

**Comment by augustinov** [comment]
It seems like there might be an issue with the contract deployment or the transaction data being sent. Here are some steps you can take to troubleshoot the problem:

Double-check the contract deployment script or the manual transaction data to ensure that the correct contract creation bytecode is being sent.

Verify that the token contract address is correct and that the contract code is deployed on the Ethereum mainnet.

Check if there are any restrictions or requirements for bridging ERC-20 tokens with the zkSync protocol, such as minimum token balances or specific token contract configurations.

Review the error messages and transaction logs for any additional details that might indicate the source of the problem.

Consider reaching out to the zkSync support team or community forums for assistance, as they may be able to provide further guidance or troubleshooting steps specific to the zkSync protocol and contract interactions.

**Comment by dawn71v** [comment]
Ensure that the deposit transaction includes the correct data and parameters required by the zkSync protocol. Double-check the token contract address and ensure that the value parameter is set correctly for token deposits. Additionally, verify that the transaction is being executed on the correct network (Mainnet) and that the zkSync node is properly synchronized.

**Comment by dunminecom** [comment]
It seems there might be an issue with the transaction execution or contract creation process when depositing STORJ tokens to create the L2 bridged contract on zkSync. The error messages indicate potential problems with the transaction execution and contract creation steps. Further investigation and debugging are needed to identify and resolve the issue.

**Comment by andrewtotta** [comment]
The user is encountering issues when attempting to deposit STORJ tokens to create an L2 bridged contract on zkSync. They have tried using the hardhat example and creating a transaction manually, but have encountered errors and issues with the contract not being created. The user is seeking guidance on the easiest way to deposit a new ERC-20 token and has provided error messages and logs. They have not made any recent changes to the contract and have not provided any relevant code.
